lib/provisioner/worker/plugins/automators/docker_automator/docker_automator.rb

```diff
@@ -114,21 +118,35 @@ def portmap
 
   def envmap
     # TODO: allow commas inside quotes
-    @fields['environment_variables'].split(',').map {|x| "-e #{x}" }.join(' ') if @fields.key?('environment_variables')
+    @envs.map {|x| "-e #{x}" }.join(' ')
   end
 
   def linkmap
-    @fields['links'].split(',').map {|x| "--link #{x}" }.join(' ') if @fields.key?('links')
+    @links.map {|x| "--link #{x}" }.join(' ')
   end
 
   def volmap
-    @fields['volumes'].split(',').map {|x| "-v #{x}" }.join(' ') if @fields.key?('volumes')
+    @vols.map {|x| "-v #{x}" }.join(' ')
   end
 
   def container_name(image_name)
     "--name coopr-#{image_name.split('/').last}"
   end
 
+  def setup_host_volumes(volumes)
+    volumes.each do |volume|
```

Comment:
if you accept my comment about initializing ``@vols`` as an array, then take out the ``.split(',')`` here too.